pkg/metabase/metabase.go

```diff
@@ -85,17 +118,31 @@ func (m *Metabase) HideOtherTables(ctx context.Context, dbID int, table string)
 		return err
 	}
 
-	other := []int{}
+	includedTables := []string{bq.Table}
+	if !isRestrictedDatabase(mbMeta) {
+		includedTables, err = m.repo.GetOpenMetabaseTablesInSameBigQueryDataset(ctx, bq.ProjectID, bq.Dataset)
+		if err != nil {
+			return err
+		}
+	}
+
+	includedIDs := []int{}
+	excludedIDs := []int{}
 	for _, t := range v.Tables {
-		if t.Name != table {
-			other = append(other, t.ID)
+		if contains(includedTables, t.Name) {
+			includedIDs = append(includedIDs, t.ID)
+		} else {
+			excludedIDs = append(excludedIDs, t.ID)
 		}
 	}
 
-	if len(other) == 0 {
-		return nil
+	if len(excludedIDs) != 0 {
+		if err := m.client.HideTables(ctx, excludedIDs); err != nil {
+			return err
+		}
 	}
-	return m.client.HideTables(ctx, other)
+
+	return m.client.ShowTables(ctx, includedIDs)
```

Comment:
Litt usikker på om vi ønsker å tvinge fram at alle åpne tabeller vises. Brukerne har tilgang til å gå til "Table metadata" og velge hvilke tabeller som skal vises. Blir fort veldig mange tabeller i noen datasett så kan hende noen ønsker at ikke alle skal være synlig for ryddighetens skyld?